### DISCUSSION TITLE
MultiQuantitativeTrack: changing track height and minScore/maxScore?

### DISCUSSION THREAD

**Discussion by lyijin** [GMOD/jbrowse-components#4167] [Q&A] [answered]
i've been testing out a MultiQuantitativeTrack to visualise 10 methylation datasets (but started with 2 for now).

i'd like to specify the rowHeight/height for the tracks, tried popping the property within the "adapter" field and outside that, didn't seem to work. i assume it's possible since #3030 was marked as closed. i read through the changes but couldn't grok the TS code very well, apologies. it'd be nice to be able to set per-subadapter height, but i'm also OK with setting per-track height and all subadapters having [overall height divided by number of subadapters].

i'd also like to set the minScore/maxScore. my data ranges from 0-100, but to force a visual change, i set the range to 0-1000 in the JSON.

neither option seems to work, have i nested the properties in the wrong field? snippet below. cheers.

```JSON
    {
      "type": "MultiQuantitativeTrack",
      "trackId": "b_5Nv5C",
      "name": "5N v 5C",
      "category": ["b methylome"],
      "assemblyNames": ["grch38p13_lambda_puc"],
      "minScore": 0,      <-- is this where this should go?
      "maxScore": 1000,   <-- is this where this should go?
      "rowHeight": 1000,  <-- is this where this should go?
      "adapter": {
        "type": "MultiWiggleAdapter",
        "subadapters": [
          {
            "type": "BigWigAdapter",
            "name": "N01",
            "bigWigLocation": {"uri": "b_beta/N01.filt.bw"},
            "group": "Normal squamous",
            "color": "#984ea3"
          },
          {
            "type": "BigWigAdapter",
            "name": "C02",
            "bigWigLocation": {"uri": "b_beta/C02.filt.bw"},
            "group": "Cancer",
            "color": "#e41a1c"
          }
        ]
      }
    },
```

these two properties can be set manually in the browser, but was hoping that i can control those via JSON directly. cheers.

**Comment by cmdcolin** [comment, marked as answer]
currently it can't be set on a per subtrack basis, it does try to autofit into the track. but, here is an example of setting height of multiwiggle, it is set on the "display" which is a bit tricky

what i often do is clone a track ("Copy track") which creates a session track, then edit it's settings, and then go to "About track" and then "Copy track config" to get a configured version of the track

you can also check the auto-generated docs for the available slots

https://jbrowse.org/jb2/docs/config/multilinearwiggledisplay/ and also https://jbrowse.org/jb2/docs/config/sharedwiggledisplay/ because the 'multi linear wiggle display' extends the 'shared wiggle display' :)

I know that configuring the "displays" instead of the tracks is a bit hard, we are brainstorming ways to make it easier still

```
{
  "type": "MultiQuantitativeTrack",
  "trackId": "mytrack",
  "name": "MultiWig",
  "assemblyNames": [ "hg19"],
  "adapter": {
    "type": "MultiWiggleAdapter",
    "bigWigs": [
      "https://www.encodeproject.org/files/ENCFF055ZII/@@download/ENCFF055ZII.bigWig",
      "https://www.encodeproject.org/files/ENCFF826HEW/@@download/ENCFF826HEW.bigWig",
      "https://www.encodeproject.org/files/ENCFF858LIM/@@download/ENCFF858LIM.bigWig",
      "https://www.encodeproject.org/files/ENCFF425TNW/@@download/ENCFF425TNW.bigWig",
      "https://www.encodeproject.org/files/ENCFF207RBY/@@download/ENCFF207RBY.bigWig",
      "https://www.encodeproject.org/files/ENCFF289CTN/@@download/ENCFF289CTN.bigWig",
      "https://www.encodeproject.org/files/ENCFF884IEG/@@download/ENCFF884IEG.bigWig",
      "https://www.encodeproject.org/files/ENCFF495SBQ/@@download/ENCFF495SBQ.bigWig",
      "https://www.encodeproject.org/files/ENCFF959EZF/@@download/ENCFF959EZF.bigWig",
      "https://www.encodeproject.org/files/ENCFF926YZX/@@download/ENCFF926YZX.bigWig",
      "https://www.encodeproject.org/files/ENCFF269CHA/@@download/ENCFF269CHA.bigWig",
      "https://www.encodeproject.org/files/ENCFF857KTJ/@@download/ENCFF857KTJ.bigWig",
      "https://www.encodeproject.org/files/ENCFF109KCQ/@@download/ENCFF109KCQ.bigWig",
      "https://www.encodeproject.org/files/ENCFF942TZX/@@download/ENCFF942TZX.bigWig",
      "https://www.encodeproject.org/files/ENCFF140HPM/@@download/ENCFF140HPM.bigWig",
      "https://www.encodeproject.org/files/ENCFF305JRR/@@download/ENCFF305JRR.bigWig",
      "https://www.encodeproject.org/files/ENCFF739FDJ/@@download/ENCFF739FDJ.bigWig",
      "https://www.encodeproject.org/files/ENCFF518OJP/@@download/ENCFF518OJP.bigWig",
      "https://www.encodeproject.org/files/ENCFF810HHS/@@download/ENCFF810HHS.bigWig",
      "https://www.encodeproject.org/files/ENCFF939JSB/@@download/ENCFF939JSB.bigWig",
      "https://www.encodeproject.org/files/ENCFF041TAK/@@download/ENCFF041TAK.bigWig"
    ]
  },
  "displays": [
    {
      "type": "MultiLinearWiggleDisplay",
      "displayId": "mytrack-MultiLinearWiggleDisplay",
      "height": 800 <-- this is the configuration of the height
      "minScore": 0,
      "maxScore": 100
    }
  ]
}
```

**Reply by cmdcolin** [reply]
the concepts of the displays exists because the same track can be "displayed" in different view types e.g. dotplot or circular, so the options in one might not make sense for the other, but it does complicate where the admin places the config!

**Reply by lyijin** [reply]
excellent, can confirm things now work as intended. thanks for explaining the "displays" concept.

oh i did discover the "copy config" button (the one that happily exclaims "copied to clipboard!" when clicked, right?). the strange thing was, after changing the height and min/maxScores, when i pasted the config into a text editor, the display part was just

```
  "displays": [
    {
      "type": "MultiLinearWiggleDisplay",
      "displayId": "b_5Nv5C-MultiLinearWiggleDisplay-1705383697297"
    }
  ]
```
(and this was before knowing where to put height and min/maxScores.)

now that i've specified height & min/maxScores in my track (600, 0, 100 respectively), i cloned the track, dragged out the height and changed max to 250. when i copied the config from this track, what i see copied out are the pre-configured values--not the currently displayed ones. which was why i wasn't able to figure out where things should go from trial-and-error initially!

```
  "displays": [
    {
      "type": "MultiLinearWiggleDisplay",
      "displayId": "b_5Nv5C-MultiLinearWiggleDisplay-1705384502182",
      "height": 600,
      "minScore": 0,
      "maxScore": 100
    }
  ]
```
